plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoTypeMapping.java

```diff
@@ -311,6 +311,15 @@ public void testArrayNulls()
         }
     }
 
+    @Test
+    public void testJSON()
+    {
+        // Verify only SELECT instead of using SqlDataTypeTest to avoid MongoDB json cast failures
```

Comment:
`SqlDataTypeTest` with `trinoCreateAsSelect` and `trinoCreateAndInsert` works well as far as I confirmed. What error did you face?

Reply:
I wrote the test case like this in the first place:
```
        SqlDataTypeTest.create()
                .addRoundTrip("json", "'{\"id\":0,\"name\":\"user_0\"}'", JSON, "'{\"id\":0,\"name\":\"user_0\"}'")
                .execute(getQueryRunner(), trinoCreateAsSelect("test_json"))
                .execute(getQueryRunner(), trinoCreateAndInsert("test_json"));
```
This is the generated SQL:
```
CREATE TABLE test_jsonjxn0mmcfdl AS SELECT CAST('{"id":0,"name":"user_0"}' AS json) col_0
```
And it failed with reason:
```
io.trino.spi.TrinoException: Can't convert json to MongoDB Document: "{\"id\":0,\"name\":\"user_0\"}"
	at io.trino.plugin.mongodb.MongoPageSink.getObjectValue(MongoPageSink.java:188)
	at io.trino.plugin.mongodb.MongoPageSink.appendPage(MongoPageSink.java:112)
	at io.trino.operator.TableWriterOperator.addInput(TableWriterOperator.java:257)
	at io.trino.operator.Driver.processInternal(Driver.java:415)
	at io.trino.operator.Driver.lambda$process$10(Driver.java:313)
	at io.trino.operator.Driver.tryWithLock(Driver.java:703)
	at io.trino.operator.Driver.process(Driver.java:305)
	at io.trino.operator.Driver.processForDuration(Driver.java:276)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:737)
	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:164)
	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:490)
	at io.trino.$gen.Trino_testversion____20220821_012151_71.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.bson.BsonInvalidOperationException: readStartDocument can only be called when CurrentBSONType is DOCUMENT, not when CurrentBSONType is STRING.
	at org.bson.AbstractBsonReader.verifyBSONType(AbstractBsonReader.java:689)
	at org.bson.AbstractBsonReader.checkPreconditions(AbstractBsonReader.java:721)
	at org.bson.AbstractBsonReader.readStartDocument(AbstractBsonReader.java:449)
	at org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:174)
	at org.bson.codecs.DocumentCodec.decode(DocumentCodec.java:44)
	at org.bson.Document.parse(Document.java:126)
	at org.bson.Document.parse(Document.java:111)
	at io.trino.plugin.mongodb.MongoPageSink.getObjectValue(MongoPageSink.java:185)
	... 14 more
```

Reply:
Thanks for sharing the example. Prepending `json ` in literals makes the test work. 
```java
.addRoundTrip("json", "json '{\"id\":0,\"name\":\"user_0\"}'", JSON, "json '{\"id\":0,\"name\":\"user_0\"}'")
```

Reply:
> Thanks for sharing the example. Prepending `json ` in literals makes the test work.
> 
> ```java
> .addRoundTrip("json", "json '{\"id\":0,\"name\":\"user_0\"}'", JSON, "json '{\"id\":0,\"name\":\"user_0\"}'")
> ```

Yes, this totally works as expected, thanks.

Reply:
Case 1: the original issue, https://github.com/trinodb/trino/issues/13536

Case 2: when case 1 is solved, it only means that a json filter would not cause errors when building the query, but the filter may be eliminated entirely(which is not correct, so a new test should be fired to ensure this filter is still working)

the previous test which used only "SELECT": solves both case 1 and 2 (2 rows are inserted and only 1 row is back)
SqlDataTypeTest: solves only case 1 (at lease for now)

Reply:
`SqlDataTypeTest` has `verifyPredicate`. Note that the current `SELECT` base doesn't cover whether the predicate is pushed down or not either.

Reply:
OK, code changed.

I've seen the intrinsic logics of `SqlDataTypeTest`, they are very marvelous ~~